activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -81,6 +81,10 @@ def version
         @version ||= Version.new(full_version.match(/^\d+\.\d+\.\d+/)[0])
       end
 
+      def mariadb? # :nodoc:
+        full_version =~ /mariadb/i
+      end
```

Comment:
Note: Moved this. This is similar to  doing https://github.com/vipulnsward/rails/commit/d9e74ace9f0d0bba39c3b836b3d79b059d516eb0